harness/determined/core/_distributed.py

```diff
@@ -68,7 +68,7 @@ def __init__(
         self._is_chief = self.rank == 0
         self._is_local_chief = self.local_rank == 0
 
-        if self.cross_size > 1:
+        if self.local_size != self.size:
```

Comment:
The AssertionError should maybe say "This a distributed job (`local_size != size`) but chief_ip was not provided ..."